server/apps/packages/package_service.py

```diff
@@ -272,6 +272,19 @@ def replace_ref_in_package(self, package, old_ref_id, new_ref_id):
                 g[ITEM_REF] = new_ref_id
                 g['guid'] = new_ref_id
 
+    def update_field_in_package(self, package, ref_id, field, field_value):
+        """
+        Locates the reference with the ref_id and replaces field value
+        :param package: Package
+        :param ref_id: reference id
+        :param field: field to be replaced
+        :param field_value: value to be used
+        """
+        non_root_groups = (group for group in package.get(GROUPS, []) if group.get(GROUP_ID) != ROOT_GROUP)
```

Comment:
Thanks for your comment Ioan. I agree with Peter that in this instance it would be easier for me to read it if I keep the comprehension separate.